Source/SwiftLintFramework/Protocols/Rule.swift

```diff
@@ -32,6 +32,12 @@ public protocol ConfigurableRule: Rule {
     func isEqualTo(rule: ConfigurableRule) -> Bool
 }
 
+public protocol ViolationLevelRule: ConfigurableRule {
+    var warning: RuleParameter<Int> { get set }
+    var error: RuleParameter<Int> { get set }
+}
+
+@available(*, deprecated=0.5.6, message="Use ConfigurableRule instead.")
```

Comment:
Got it. It was more for myself to practice using that attribute and flag warnings for visibility. Planning on removing.